pkg/graph/resolvers/key_contact_test.go

```diff
@@ -0,0 +1,291 @@
+	newCategory, err := CreateKeyContactCategory(
+		suite.testConfigs.Logger,
+		suite.testConfigs.Principal,
+		suite.testConfigs.Store,
+		categoryName,
+	)
+	suite.NoError(err)
+
+	userName := "John Doe"
+	subjectArea := "Healthcare"
+	subjectCategoryID := newCategory.ID
+
+	contact, err := CreateKeyContactUser(
+		suite.testConfigs.Context,
+		suite.testConfigs.Logger,
+		suite.testConfigs.Principal,
+		suite.testConfigs.Store,
+		nil,
+		nil,
+		email.AddressBook{},
+		userName,
+		subjectArea,
+		subjectCategoryID,
+		userhelpers.GetUserInfoAccountInfoWrapperFunc(suite.stubFetchUserInfo),
+	)
+
+	suite.NoError(err)
+	suite.NotNil(contact)
+	if suite.NotNil(contact.UserID) {
+		userAccount, err := userhelpers.UserAccountGetByIDLOADER(suite.testConfigs.Context, *contact.UserID)
+		suite.NoError(err)
+		if suite.NotNil(userAccount.Username) {
+			suite.Equal(userName, *userAccount.Username)
+		}
+	}
+	suite.Equal(subjectArea, contact.SubjectArea)
+	suite.Equal(subjectCategoryID, contact.SubjectCategoryID)
+}
+
+// TestCreateKeyContactMailbox validates creating a mailbox contact for key contact.
+func (suite *ResolverSuite) TestCreateKeyContactMailbox() {
+	categoryName := "Test Category"
+	newCategory, err := CreateKeyContactCategory(
+		suite.testConfigs.Logger,
+		suite.testConfigs.Principal,
+		suite.testConfigs.Store,
+		categoryName,
+	)
+	suite.NoError(err)
+
+	mailboxTitle := "Support Team"
+	mailboxAddress := "support@example.com"
+	subjectArea := "Healthcare"
+	subjectCategoryID := newCategory.ID
+
+	contact, err := CreateKeyContactMailbox(
+		suite.testConfigs.Context,
+		suite.testConfigs.Logger,
+		suite.testConfigs.Principal,
+		suite.testConfigs.Store,
+		nil,
+		nil,
+		email.AddressBook{},
+		mailboxTitle,
+		mailboxAddress,
+		subjectArea,
+		subjectCategoryID,
+	)
+
+	suite.NoError(err)
+	suite.NotNil(contact)
+	suite.Equal(mailboxTitle, *contact.Name)
+	suite.Equal(mailboxAddress, *contact.Email)
+	suite.Equal(subjectArea, contact.SubjectArea)
+	suite.Equal(subjectCategoryID, contact.SubjectCategoryID)
+}
+
+// TestGetKeyContactByIDLOADER validates fetching a key contact by its ID using the loader.
+func (suite *ResolverSuite) TestGetKeyContactByIDLOADER() {
+	categoryName := "Test Category"
+	newCategory, err := CreateKeyContactCategory(
+		suite.testConfigs.Logger,
+		suite.testConfigs.Principal,
+		suite.testConfigs.Store,
+		categoryName,
+	)
+	suite.NoError(err)
+
+	// Create a mailbox contact using the existing method
+	mailboxTitle := "Support Team"
+	mailboxAddress := "support@example.com"
+	subjectArea := "Healthcare"
+	subjectCategoryID := newCategory.ID
+
+	contact, err := CreateKeyContactMailbox(
+		suite.testConfigs.Context,
+		suite.testConfigs.Logger,
+		suite.testConfigs.Principal,
+		suite.testConfigs.Store,
+		nil,
+		nil,
+		email.AddressBook{},
+		mailboxTitle,
+		mailboxAddress,
+		subjectArea,
+		subjectCategoryID,
+	)
+
+	suite.NoError(err)
+
+	// Fetch key contact using the loader
```

Comment:
This loader test still needs to be addressed. This test is good because you are validating the resolver, but we still would ideally have the loader for get by ID tested as well (perhaps a separate unit test?).